src/java.base/share/classes/sun/security/tools/keytool/Main.java

```diff
@@ -2073,6 +2062,37 @@ private void doGenKeyPair(String alias, String dname, String keyAlgName,
         keyStore.setKeyEntry(alias, privKey, keyPass, finalChain);
     }
 
+    /**
+     * Returns the full display name of the given key object. Could be
+     * - "X25519", if its getParams() is NamedParameterSpec
+     * - "EC (secp256r1)", if it's an EC key
+     * - "1024-bit RSA", other known keys
+     * - plain algorithm name, otherwise
+     *
+     * Note: the same method appears in keytool and jarsigner which uses
+     * same resource string defined in their own Resources.java.
+     *
```

Comment:
This method appears in both keytool and jarsigner as documented here. Can we define a common method in KeyUtil class, which can return "size" and "alg", in order for its caller to construct its display?

Reply:
I prefer to consolidate the logic in a single location. A record containing `size` and `alg` doesn’t seem to carry clear semantic meaning. Additionally, the `alg` field here has already been modified and does not solely represent the algorithm name.